client/src/components/categorical/occupancy.js

```diff
@@ -2,60 +2,172 @@
 import React from "react";
 import { connect } from "react-redux";
 import * as d3 from "d3";
+import {
+  Popover,
+  PopoverInteractionKind,
+  Position,
+  Classes
+} from "@blueprintjs/core";
 
 @connect()
 class Occupancy extends React.Component {
-  render() {
-    const { occupancy, colorScale, colorAccessor, schema, world } = this.props;
-    const width = 100;
-    const height = 11;
+  _WIDTH = 100;
 
-    const categories = schema.annotations.obsByName[colorAccessor]?.categories;
+  _HEIGHT = 11;
+
+  createHistogram = () => {
```

Comment:
general comment - all changes lack comments.  Please comment sufficiently that someone else can read and understand the code quickly (never having seen it before).